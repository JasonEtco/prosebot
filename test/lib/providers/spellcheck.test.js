const SpellCheck = require('../../../lib/providers/spellcheck')

describe('SpellCheck provider', () => {
  let provider

  beforeEach(() => {
    const map = new Map()

    const obj = {
      'filename.md': 'iam not a wurd',
      'anotherfile.md': 'Tenis is a fun sporq',
      'gibberish.md': 'AbsdfsdalkjhfthjsdfsdfAbsdfsdalkjhfthjsdfsdfAbsdfs',
      'technical-doc.md':
        'This is a `config.yml` file. Here is some code: ```js\nconsole.log(pizza_slice)\n```',
    }

    for (const key in obj) {
      map.set(key, obj[key])
    }

    provider = new SpellCheck(map)
    provider.suggestCorrections = jest.fn(() =>
      Promise.resolve(['correctly', 'spelled', 'words'])
    )
  })

  describe('#buildReasonString', () => {
    it('returns the expected string with zero possible corrections', async () => {
      provider.suggestCorrections.mockImplementation(() => Promise.resolve([]))

      const actual = await provider.buildResults()

      expect(actual.get('filename.md')).toMatchSnapshot()
    })

    it('returns the expected string with some corrections', async () => {
      const actual = await provider.buildResults()

      expect(actual.get('filename.md')).toMatchSnapshot()
    })
  })

  describe('#buildResults', () => {
    it('creates the file map with appropriate reasons', async () => {
      const actual = await provider.buildResults()

      expect(actual).toMatchSnapshot()
    })
  })
})
