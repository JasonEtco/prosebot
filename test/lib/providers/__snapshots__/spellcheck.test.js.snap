// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellCheck provider #buildReasonString returns the expected string with some corrections 1`] = `
Array [
  Object {
    "line": 1,
    "reason": "\\"iam\\" is misspelled. How about: correctly, spelled, words",
  },
  Object {
    "line": 1,
    "reason": "\\"wurd\\" is misspelled. How about: correctly, spelled, words",
  },
]
`;

exports[`SpellCheck provider #buildReasonString returns the expected string with zero possible corrections 1`] = `
Array [
  Object {
    "line": 1,
    "reason": "\\"iam\\" is misspelled. ",
  },
  Object {
    "line": 1,
    "reason": "\\"wurd\\" is misspelled. ",
  },
]
`;

exports[`SpellCheck provider #buildResults creates the file map with appropriate reasons 1`] = `
Map {
  "filename.md" => Array [
    Object {
      "line": 1,
      "reason": "\\"wurd\\" is misspelled. How about: correctly, spelled, words",
    },
    Object {
      "line": 1,
      "reason": "\\"iam\\" is misspelled. How about: correctly, spelled, words",
    },
  ],
  "anotherfile.md" => Array [
    Object {
      "line": 1,
      "reason": "\\"Tenis\\" is misspelled. How about: correctly, spelled, words",
    },
    Object {
      "line": 1,
      "reason": "\\"sporq\\" is misspelled. How about: correctly, spelled, words",
    },
  ],
  "gibberish.md" => Array [
    Object {
      "line": 1,
      "reason": "\\"AbsdfsdalkjhfthjsdfsdfAbsdfsdalkjhfthjsdfsdfAbsdfs\\" is misspelled. How about: correctly, spelled, words",
    },
  ],
  "technical-doc.md" => Array [],
}
`;
